homeassistant/components/tuneblade_remote/media_player.py

```diff
@@ -0,0 +1,161 @@
+"""TuneBlade Remote media player entity."""
+
+import asyncio
+import logging
+from typing import Any
+
+from homeassistant.components.media_player import (
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN, MASTER_ID
+from .coordinator import TuneBladeDataUpdateCoordinator
+from .entity import TuneBladeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up TuneBlade Remote from a config entry."""
+    coordinator: TuneBladeDataUpdateCoordinator = config_entry.runtime_data
+
+    runtime_data = hass.data.setdefault(DOMAIN, {}).setdefault(
+        config_entry.entry_id, {}
+    )
+    added_ids = runtime_data.setdefault("added_ids", set())
+
+    entities_master = []
+    entities_others = []
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id == MASTER_ID:
+            entities_master.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id != MASTER_ID:
+            entities_others.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    if entities_master:
+        async_add_entities(entities_master, True)
+
+    if entities_others:
+        async_add_entities(entities_others, True)
```

Comment:
1. We should only call `async_add_entities` once
2. Can we use `main` as wording?
3. I think we can merge these loops
4. We don't need to call `async_add_entities` with `True` as they already have a state, so we don't need to fetch a new one